sdk/core/azure-core/test/ut/curl_session_test.cpp

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "curl_session.hpp"
+
+#include <azure/core/http/curl/curl.hpp>
+#include <azure/core/http/http.hpp>
+
+using ::testing::_;
+using ::testing::DoAll;
+using ::testing::Return;
+using ::testing::SetArrayArgument;
+
+namespace Azure { namespace Core { namespace Test {
+
+  TEST_F(CurlSession, successCall)
+  {
+    std::string response(
+        "HTTP/1.1 200 Ok\r\nContent-Type: text/html; charset=UTF-8\r\n\r\n{\r\n\"somejson\":45\r}");
+
+    // Can't mock the curMock directly from a unique ptr, heap allocate it first and then make a
+    // unique ptr for it
+    MockCurlNetworkConnection* curlMock = new MockCurlNetworkConnection();
```

Comment:
This line of code leaks.  You need to delete the memory you allocate when you are done with it

Reply:
gmock is intelligent and tells you if there is anything leak at the end of the test.

On line 31, We create a unique_ptr from this pointer to take care about deleting the memory.

I also add a mock destructor to make sure we are destructing the object :)